graylog2-web-interface/src/routing/loadAsync.tsx

```diff
@@ -14,22 +14,60 @@
  * along with this program. If not, see
  * <http://www.mongodb.com/licensing/server-side-public-license>.
  */
-import React from 'react';
+import * as React from 'react';
+import { Suspense } from 'react';
 import PropTypes from 'prop-types';
-import loadable from 'loadable-components';
+
+import Spinner from 'components/common/Spinner';
+
+type Error = {
+  message: string;
+};
 
 type Props = {
-  error: {
-    message: string;
-  };
+  error: Error;
 };
 
-const ErrorComponent: React.FC<Props> = ({ error }: Props) => <div>Loading component failed: {error.message}</div>;
+const ErrorComponent: React.ComponentType<Props> = ({ error }: Props) => <div>Loading component failed: {error.message}</div>;
```

Comment:
Isn't this a `React.FunctionComponent`?  Why did you change it into a `React.ComponentType`?

Reply:
`React.ComponentType` is the abstraction, it covers both class and functional components. We do not really care in this case if it's a class or functional component, so I captured that in its type.

Reply:
Makes sense! Thanks for clarifying, never used / seen it before.